[Revert]: (253931@main): [ macOS wk1 Debug ] media/track/track-description-cue.html consistently hits ASSERTION FAILED: !m_currentSpeechUtterance

karlrackler opened the pull request:
#### 3bd99112f8a9d51a38a5880d7f51fed3f73bafc1
<pre>
[Revert]: (253931@main): [ macOS wk1 Debug ] media/track/track-description-cue.html consistently hits ASSERTION FAILED: !m_currentSpeechUtterance
<a href="https://bugs.webkit.org/show_bug.cgi?id=244683">https://bugs.webkit.org/show_bug.cgi?id=244683</a>
rdar://problem/99449962

Unreviewed revert
This reverts 253931@main because it consistently hits ASSERTION FAILED: !m_currentSpeechUtterance.

* LayoutTests/TestExpectations:
* LayoutTests/media/track/captions-webvtt/captions-descriptions.vtt: Removed.
* LayoutTests/media/track/track-description-cue-expected.txt: Removed.
* LayoutTests/media/track/track-description-cue.html: Removed.
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/mac/TestExpectations:
* Source/WTF/Scripts/Preferences/WebPreferencesExperimental.yaml:
* Source/WebCore/Modules/speech/SpeechSynthesis.cpp:
(WebCore::SpeechSynthesis::fireEvent const):
(WebCore::SpeechSynthesis::fireErrorEvent const):
(WebCore::SpeechSynthesis::handleSpeakingCompleted):
(WebCore::SpeechSynthesis::boundaryEventOccurred):
(WebCore::SpeechSynthesis::didStartSpeaking):
(WebCore::SpeechSynthesis::didPauseSpeaking):
(WebCore::SpeechSynthesis::didResumeSpeaking):
* Source/WebCore/Modules/speech/SpeechSynthesis.h:
(WebCore::SpeechSynthesis::userGestureRequiredForSpeechStart const):
(WebCore::SpeechSynthesis::removeBehaviorRestriction):
* Source/WebCore/Modules/speech/SpeechSynthesisUtterance.cpp:
(WebCore::SpeechSynthesisUtterance::create):
(WebCore::SpeechSynthesisUtterance::SpeechSynthesisUtterance):
(WebCore::SpeechSynthesisUtterance::eventOccurred): Deleted.
(WebCore::SpeechSynthesisUtterance::errorEventOccurred): Deleted.
* Source/WebCore/Modules/speech/SpeechSynthesisUtterance.h:
(): Deleted.
* Source/WebCore/Modules/speech/SpeechSynthesisUtterance.idl:
* Source/WebCore/html/HTMLMediaElement.cpp:
(WebCore::m_categoryAtMostRecentPlayback):
(WebCore::HTMLMediaElement::updateActiveTextTrackCues):
(WebCore::HTMLMediaElement::seekWithTolerance):
(WebCore::HTMLMediaElement::seekTask):
(WebCore::HTMLMediaElement::finishSeek):
(WebCore::HTMLMediaElement::addTextTrack):
(WebCore::HTMLMediaElement::configureTextTrackGroup):
(WebCore::HTMLMediaElement::playPlayer):
(WebCore::HTMLMediaElement::pausePlayer):
(WebCore::HTMLMediaElement::captionPreferencesChanged):
(WebCore::HTMLMediaElement::effectiveVolume const):
(WebCore::m_logIdentifier): Deleted.
* Source/WebCore/html/HTMLMediaElement.h:
* Source/WebCore/html/shadow/MediaControlTextTrackContainerElement.cpp:
(WebCore::MediaControlTextTrackContainerElement::updateDisplay):
* Source/WebCore/html/track/InbandGenericTextTrack.cpp:
* Source/WebCore/html/track/TextTrack.cpp:
(WebCore::TextTrack::trackIndex):
(WebCore::TextTrack::isRendered):
(WebCore::TextTrack::trackIndexRelativeToRenderedTracks):
(WebCore::TextTrack::isSpoken): Deleted.
(WebCore::TextTrack::speechSynthesis): Deleted.
* Source/WebCore/html/track/TextTrack.h:
* Source/WebCore/html/track/TextTrackCue.cpp:
(WebCore::operator&lt;&lt;):
* Source/WebCore/html/track/TextTrackCue.h:
(WebCore::TextTrackCue::text const): Deleted.
(WebCore::TextTrackCue::speak): Deleted.
* Source/WebCore/html/track/VTTCue.cpp:
(WebCore::VTTCue::updateDisplayTree):
(WebCore::VTTCue::getDisplayTree):
(WebCore::VTTCue::toJSON const):
(WebCore::VTTCue::speak): Deleted.
* Source/WebCore/html/track/VTTCue.h:
(WebCore::VTTCue::text const):
(WebCore::VTTCue::speechUtterance const): Deleted.
* Source/WebCore/html/track/VTTCue.idl:
* Source/WebCore/page/CaptionUserPreferences.cpp:
(WebCore::CaptionUserPreferences::userPrefersTextDescriptions const):
(WebCore::CaptionUserPreferences::textTrackSelectionScore const):
* Source/WebCore/page/CaptionUserPreferencesMediaAF.cpp:
(WebCore::CaptionUserPreferencesMediaAF::userPrefersCaptions const):
(WebCore::CaptionUserPreferencesMediaAF::userPrefersTextDescriptions const): Deleted.
* Source/WebCore/page/CaptionUserPreferencesMediaAF.h:
* Source/WebCore/platform/cf/MediaAccessibilitySoftLink.cpp:
* Source/WebCore/platform/cf/MediaAccessibilitySoftLink.h:
* Source/WebCore/platform/cocoa/PlatformSpeechSynthesizerCocoa.mm:
(-[WebSpeechSynthesisWrapper speakUtterance:]):
* Source/WebCore/platform/graphics/InbandGenericCue.cpp:
(WebCore::InbandGenericCue::toJSONString const):
* Source/WebCore/platform/graphics/iso/ISOVTTCue.cpp:
(WebCore::ISOWebVTTCue::toJSONString const):
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::setMediaElementVolumeLocked):
(WebCore::Internals::speechSynthesisUtteranceForCue): Deleted.
(WebCore::Internals::mediaElementCurrentlySpokenCue): Deleted.
* Source/WebCore/testing/Internals.h:
* Source/WebCore/testing/Internals.idl:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::speechSynthesisSpeak):

Canonical link: <a href="https://commits.webkit.org/254241@main">https://commits.webkit.org/254241@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/47988fe7491da0e779ab397599e42e69c5dfed43

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/88487 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/33008 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/19351 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/97663 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/153142 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/92449 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/64/builds/31503 "Built successfully") | [  ~~🛠 mac-debug~~](https://ews-build.webkit.org/#/builders/71/builds/27106 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/80680 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/92315 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/94090 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/68/builds/25017 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/75395 "Built successfully") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/24982 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/79905 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/80003 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/67945 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/80209 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/29120 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/13993 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/73978 "Built successfully") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/67/builds/29084 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/70/builds/15011 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/26220 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/2983 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/32342 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/37937 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/76834 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/31167 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/62/builds/34120 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/17044 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

eric-carlson commented on Source/WebCore/Modules/speech/SpeechSynthesisUtterance.h:
```diff
@@ -30,19 +30,20 @@
 class WEBCORE_EXPORT SpeechSynthesisUtterance final : public PlatformSpeechSynthesisUtteranceClient, public RefCounted<SpeechSynthesisUtterance>, public EventTarget {
+=======
+class SpeechSynthesisUtterance final : public PlatformSpeechSynthesisUtteranceClient, public RefCounted<SpeechSynthesisUtterance>, public EventTargetWithInlineData {
+>>>>>>> parent of 969ac99ba7af (AX: Consider VTT-based audio descriptions with text-to-speech.)
```
I think this should be:

```
class SpeechSynthesisUtterance final : public PlatformSpeechSynthesisUtteranceClient, public RefCounted<SpeechSynthesisUtterance>, public EventTargetWithInlineData {
```

eric-carlson commented on Source/WebCore/html/HTMLMediaElement.cpp:
```diff
@@ -2043,6 +2047,8 @@ void HTMLMediaElement::executeCueEnterOrLeaveAction(TextTrackCue& cue, CueAction
 }
 
+=======
+>>>>>>> parent of 969ac99ba7af (AX: Consider VTT-based audio descriptions with text-to-speech.)
```
The entire `HTMLMediaElement::setSpeechSynthesisState` method should be deleted.

eric-carlson reviewed (approved):
Thanks Karl!

eric-carlson commented on Source/WebCore/Modules/speech/SpeechSynthesisUtterance.h:
```diff
@@ -30,19 +30,16 @@
 namespace WebCore {
 
-class WEBCORE_EXPORT SpeechSynthesisUtterance final : public PlatformSpeechSynthesisUtteranceClient, public RefCounted<SpeechSynthesisUtterance>, public EventTarget {
+class SpeechSynthesisUtterance final : public PlatformSpeechSynthesisUtteranceClient, public RefCounted<SpeechSynthesisUtterance>, public EventTargetWithInlineData {
```
I was wrong before, I believe this should derive from `EventTarget` instead of `EventTargetWithInlineData`

eric-carlson reviewed (approved):
r=me

karlrackler committed:
[Revert]: (253931@main): [ macOS wk1 Debug ] media/track/track-description-cue.html consistently hits ASSERTION FAILED: !m_currentSpeechUtterance
https://bugs.webkit.org/show_bug.cgi?id=244683
rdar://problem/99449962

Unreviewed revert
This reverts 253931@main because it consistently hits ASSERTION FAILED: !m_currentSpeechUtterance.

* LayoutTests/TestExpectations:
* LayoutTests/media/track/captions-webvtt/captions-descriptions.vtt: Removed.
* LayoutTests/media/track/track-description-cue-expected.txt: Removed.
* LayoutTests/media/track/track-description-cue.html: Removed.
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/mac/TestExpectations:
* Source/WTF/Scripts/Preferences/WebPreferencesExperimental.yaml:
* Source/WebCore/Modules/speech/SpeechSynthesis.cpp:
(WebCore::SpeechSynthesis::fireEvent const):
(WebCore::SpeechSynthesis::fireErrorEvent const):
(WebCore::SpeechSynthesis::handleSpeakingCompleted):
(WebCore::SpeechSynthesis::boundaryEventOccurred):
(WebCore::SpeechSynthesis::didStartSpeaking):
(WebCore::SpeechSynthesis::didPauseSpeaking):
(WebCore::SpeechSynthesis::didResumeSpeaking):
* Source/WebCore/Modules/speech/SpeechSynthesis.h:
(WebCore::SpeechSynthesis::userGestureRequiredForSpeechStart const):
(WebCore::SpeechSynthesis::removeBehaviorRestriction):
* Source/WebCore/Modules/speech/SpeechSynthesisUtterance.cpp:
(WebCore::SpeechSynthesisUtterance::create):
(WebCore::SpeechSynthesisUtterance::SpeechSynthesisUtterance):
(WebCore::SpeechSynthesisUtterance::eventOccurred): Deleted.
(WebCore::SpeechSynthesisUtterance::errorEventOccurred): Deleted.
* Source/WebCore/Modules/speech/SpeechSynthesisUtterance.h:
(): Deleted.
* Source/WebCore/Modules/speech/SpeechSynthesisUtterance.idl:
* Source/WebCore/html/HTMLMediaElement.cpp:
(WebCore::m_categoryAtMostRecentPlayback):
(WebCore::HTMLMediaElement::updateActiveTextTrackCues):
(WebCore::HTMLMediaElement::seekWithTolerance):
(WebCore::HTMLMediaElement::seekTask):
(WebCore::HTMLMediaElement::finishSeek):
(WebCore::HTMLMediaElement::addTextTrack):
(WebCore::HTMLMediaElement::configureTextTrackGroup):
(WebCore::HTMLMediaElement::playPlayer):
(WebCore::HTMLMediaElement::pausePlayer):
(WebCore::HTMLMediaElement::captionPreferencesChanged):
(WebCore::HTMLMediaElement::effectiveVolume const):
(WebCore::m_logIdentifier): Deleted.
* Source/WebCore/html/HTMLMediaElement.h:
* Source/WebCore/html/shadow/MediaControlTextTrackContainerElement.cpp:
(WebCore::MediaControlTextTrackContainerElement::updateDisplay):
* Source/WebCore/html/track/InbandGenericTextTrack.cpp:
* Source/WebCore/html/track/TextTrack.cpp:
(WebCore::TextTrack::trackIndex):
(WebCore::TextTrack::isRendered):
(WebCore::TextTrack::trackIndexRelativeToRenderedTracks):
(WebCore::TextTrack::isSpoken): Deleted.
(WebCore::TextTrack::speechSynthesis): Deleted.
* Source/WebCore/html/track/TextTrack.h:
* Source/WebCore/html/track/TextTrackCue.cpp:
(WebCore::operator<<):
* Source/WebCore/html/track/TextTrackCue.h:
(WebCore::TextTrackCue::text const): Deleted.
(WebCore::TextTrackCue::speak): Deleted.
* Source/WebCore/html/track/VTTCue.cpp:
(WebCore::VTTCue::updateDisplayTree):
(WebCore::VTTCue::getDisplayTree):
(WebCore::VTTCue::toJSON const):
(WebCore::VTTCue::speak): Deleted.
* Source/WebCore/html/track/VTTCue.h:
(WebCore::VTTCue::text const):
(WebCore::VTTCue::speechUtterance const): Deleted.
* Source/WebCore/html/track/VTTCue.idl:
* Source/WebCore/page/CaptionUserPreferences.cpp:
(WebCore::CaptionUserPreferences::userPrefersTextDescriptions const):
(WebCore::CaptionUserPreferences::textTrackSelectionScore const):
* Source/WebCore/page/CaptionUserPreferencesMediaAF.cpp:
(WebCore::CaptionUserPreferencesMediaAF::userPrefersCaptions const):
(WebCore::CaptionUserPreferencesMediaAF::userPrefersTextDescriptions const): Deleted.
* Source/WebCore/page/CaptionUserPreferencesMediaAF.h:
* Source/WebCore/platform/cf/MediaAccessibilitySoftLink.cpp:
* Source/WebCore/platform/cf/MediaAccessibilitySoftLink.h:
* Source/WebCore/platform/cocoa/PlatformSpeechSynthesizerCocoa.mm:
(-[WebSpeechSynthesisWrapper speakUtterance:]):
* Source/WebCore/platform/graphics/InbandGenericCue.cpp:
(WebCore::InbandGenericCue::toJSONString const):
* Source/WebCore/platform/graphics/iso/ISOVTTCue.cpp:
(WebCore::ISOWebVTTCue::toJSONString const):
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::setMediaElementVolumeLocked):
(WebCore::Internals::speechSynthesisUtteranceForCue): Deleted.
(WebCore::Internals::mediaElementCurrentlySpokenCue): Deleted.
* Source/WebCore/testing/Internals.h:
* Source/WebCore/testing/Internals.idl:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::speechSynthesisSpeak):

Canonical link: https://commits.webkit.org/254241@main
